crash_rate_aggregates/fill_database.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python
+
+import re
+import sys
+import os
+import sys
+import numpy as np
+import dateutil.parser
+import pyspark.sql.types as types
+
+from moztelemetry.spark import get_pings, Histogram
+from datetime import datetime, date, timedelta
+
+FRACTION = 1.0
+
+# paths/dimensions within the ping to compare by
+COMPARABLE_DIMENSIONS = [
+    ("environment", "build", "version"),
+    ("environment", "build", "buildId"),
+    ("application", "channel"),
+    ("application", "name"),
+    ("environment", "system", "os", "name"),
+    ("environment", "system", "os", "version"),
+    ("environment", "build", "architecture"),
+    ("meta", "geoCountry"),
+    ("environment", "addons", "activeExperiment", "id"),
+    ("environment", "addons", "activeExperiment", "branch"),
+    ("environment", "settings", "e10sEnabled"),
+]
+
+# names of the comparable dimensions above, used as dimension names in the database
+DIMENSION_NAMES = [
+    "build_version",
+    "build_id",
+    "channel",
+    "application",
+    "os_name",
+    "os_version",
+    "architecture",
+    "country",
+    "experiment_id",
+    "experiment_branch",
+    "e10s_enabled",
+]
+
+assert len(COMPARABLE_DIMENSIONS) == len(DIMENSION_NAMES), "COMPARABLE_DIMENSIONS and DIMENSION_NAMES must match"
+
+def compare_crashes(pings, start_date, end_date, comparable_dimensions, dimension_names):
+    """Returns a PairRDD where keys are user configurations and values are Numpy arrays of the form [usage hours, main process crashes, content process crashes, plugin crashes]"""
+    def get_property(value, path):
+        for key in path:
+            if not isinstance(value, dict) or key not in value:
+                return None
+            value = value[key]
+        return value
+
+    def get_properties(ping):
+        result = {
+            key: get_property(ping, path)
+            for key, path in zip(dimension_names, comparable_dimensions)
+        }
+
+        submission_date = get_property(ping, ("meta", "submissionDate"))
+        result["submission_date"] = datetime.strptime(submission_date, "%Y%m%d").date() # convert the YYYYMMDD format to a real date
+
+        activity_date = get_property(ping, ("creationDate",))
+        try:
+            result["activity_date"] = dateutil.parser.parse(activity_date).date() # the activity date is the date portion of creationDate
+        except: # malformed date, like how sometimes seconds that are not in [0, 59]
+            result["activity_date"] = None
+
+        result["subsession_length"] = get_property(ping, ("payload", "info", "subsessionLength"))
+        result["doc_type"] = get_property(ping, ("meta", "docType"))
+
+        subprocess_crash_content = get_property(ping, ("payload", "keyedHistograms", "SUBPROCESS_CRASHES_WITH_DUMP", "content"))
+        result["subprocess_crash_content"] = 0 if subprocess_crash_content is None else Histogram("SUBPROCESS_CRASHES_WITH_DUMP", subprocess_crash_content).get_value()
```

Comment:
Let's not use Histogram; the goal is to avoid any pandas & numpy code as I recall hitting some perf issues with the telemetry aggregator.